File: backjoon/2775.cpp
#include <iostream>
using namespace std;

int main()
{
	int T, k, n, apart[20][20] = {0}; // apart[층][호]
	for (int i = 0; i <= 14; i++)			// 층
	{
		for (int j = 1; j <= 14; j++) // 호
		{
			apart[i][1] = 1; // 몇 층이든 1호는 무조건 1명 거주
			if (i == 0)
			{
				apart[i][j] = j;
				// cout << i << " " << j << " " << apart[i][j] << '\n';
			}
			else if (i > 0)
			{
				apart[i][j] = apart[i - 1][j] + apart[i][j - 1];
				// cout << i << " " << j << " " << apart[i][j] << '\n';
			}
		}
	}
	cin >> T;
	while (T--)
	{
		cin >> k >> n;
		cout << apart[k][n] << '\n';
	}

	return 0;
}

File: backjoon/2775.py
case_num = int(input())

for _ in range(case_num):
    floor = int(input())
    room_num = int(input())

    apart = [[0] * room_num for _ in range(floor + 1)]

    for i in range(room_num):
        apart[0][i] = i + 1

    for i in range(floor):
        for j in range(room_num):
            for k in range(j + 1):
                apart[i + 1][j] += apart[i][k]
    print(apart[floor][room_num - 1])
        

